File: awesome-calendar/src/main/java/com/archit/calendardaterangepicker/models/CalendarStyleAttributes.java
package com.archit.calendardaterangepicker.models;

import android.graphics.Typeface;
import android.graphics.drawable.Drawable;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;

public interface CalendarStyleAttributes {

    Typeface getFonts();

    void setFonts(@NonNull Typeface fonts);

    int getTitleColor();

    @NonNull
    Drawable getHeaderBg();

    int getWeekColor();

    int getRangeStripColor();

    int getSelectedDateCircleColor();

    int getSelectedDateColor();

    int getDefaultDateColor();

    int getDisableDateColor();

    int getRangeDateColor();

    float getTextSizeTitle();

    float getTextSizeWeek();

    float getTextSizeDate();

    boolean isShouldEnabledTime();

    void setWeekOffset(int weekOffset);

    int getWeekOffset();

    boolean isEditable();

    void setEditable(boolean editable);
}

File: awesome-calendar/src/main/java/com/archit/calendardaterangepicker/models/CalendarStyleAttributes.kt
package com.archit.calendardaterangepicker.models

import android.graphics.Typeface
import android.graphics.drawable.Drawable
import androidx.annotation.ColorInt

interface CalendarStyleAttributes {

    companion object {
        val DEFAULT_FIXED_DAYS_SELECTION = 7
    }

    enum class DateSelectionMode {
        FREE_RANGE,
        SINGLE,
        FIXED_RANGE
    }

    var fonts: Typeface?

    @get:ColorInt
    val titleColor: Int
    var headerBg: Drawable?

    @get:ColorInt
    val weekColor: Int

    @get:ColorInt
    val rangeStripColor: Int

    @get:ColorInt
    val selectedDateCircleColor: Int

    @get:ColorInt
    val selectedDateColor: Int

    @get:ColorInt
    val defaultDateColor: Int

    @get:ColorInt
    val disableDateColor: Int

    @get:ColorInt
    val rangeDateColor: Int
    val textSizeTitle: Float
    val textSizeWeek: Float
    val textSizeDate: Float
    val isShouldEnabledTime: Boolean
    var weekOffset: Int
    var isEditable: Boolean
    var dateSelectionMode: DateSelectionMode
    var fixedDaysSelectionNumber: Int
}

class InvalidCalendarAttributeException(override val message: String?) : IllegalArgumentException(message)